src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,585 @@
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance linear_ordered_comm_group.to_covariant_class (α : Type u)
+  [linear_ordered_comm_group α] :  covariant_class α α (*) (≤) :=
+{ elim := λ a b c bc, linear_ordered_comm_group.mul_le_mul_left _ _ bc a }
+
+
+variables {α : Type u} [lattice α] [comm_group α] [covariant_class α α (*) (≤)]
+
+-- Special case of Bourbaki A.VI.9 (1)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a`, `b` and `c` in `α`,
+$$c + (a ⊔ b) = (c + a) ⊔ (c + b).$$
+-/
+@[to_additive]
+lemma mul_sup_eq_mul_sup_mul (a b c : α) : c * (a ⊔ b) = (c * a) ⊔ (c * b) :=
+begin
+  refine le_antisymm _ (by simp),
+  rw [← mul_le_mul_iff_left (c⁻¹), ← mul_assoc, inv_mul_self, one_mul],
+  apply sup_le,
+  { simp },
+  { simp },
+end
+
+-- Special case of Bourbaki A.VI.9 (2)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$-(a ⊔ b)=(-a) ⊓ (-b).$$
+-/
+@[to_additive]
+lemma inv_sup_eq_inv_inf_inv (a b : α) : (a ⊔ b)⁻¹ = a⁻¹ ⊓ b⁻¹ :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { rw le_inf_iff,
+    split,
+    { rw inv_le_inv_iff, apply le_sup_left, },
+    { rw inv_le_inv_iff, apply le_sup_right, } },
+  { rw ← inv_le_inv_iff, simp,
+    split,
+    { rw ← inv_le_inv_iff, simp, },
+    { rw ← inv_le_inv_iff, simp, } }
+end
+
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$ -(a ⊓ b) = -a ⊔ -b.$$
+-/
+@[to_additive, simp]
+lemma inv_inf_eq_sup_inv (a b : α) : (a ⊓ b)⁻¹ = a⁻¹ ⊔ b⁻¹ :=
+begin
+  rw [← inv_inv (a⁻¹ ⊔ b⁻¹), inv_sup_eq_inv_inf_inv a⁻¹ b⁻¹, inv_inv, inv_inv],
+end
+
+-- Bourbaki A.VI.10 Prop 7
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$a ⊓ b + (a ⊔ b) = a + b.$$
+-/
+@[to_additive]
+lemma inf_mul_sup (a b : α) : a ⊓ b * (a ⊔ b) = a * b :=
+calc a⊓b * (a ⊔ b) = a ⊓ b * ((a * b) * (b⁻¹ ⊔ a⁻¹)) :
+  by {  rw mul_sup_eq_mul_sup_mul b⁻¹ a⁻¹ (a * b), simp,  }
+... = a⊓b * ((a * b) * (a ⊓ b)⁻¹) : by { rw [inv_inf_eq_sup_inv, sup_comm], }
+... = a * b                    : by { rw [mul_comm, inv_mul_cancel_right],  }
+
+
+/--
+Absolute value is a unary operator with properties similar to the absolute value of a real number.
+-/
+class has_abs (α : Type u) := (abs : α → α)
+local notation `|`a`|` := has_abs.abs a
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group_has_abs : has_abs (α)  := ⟨λa, a⊔a⁻¹⟩
+
+namespace lattice_ordered_comm_group
+
+-- Bourbaki A.VI.12 Definition 4
+/--
+Let `α` be a lattice ordered commutative group with identity `0`. For an element `a` of type `α`,
+the element `a ⊔ 0` is said to be the *positive component* of `a`, denoted `a⁺`.
+-/
+@[to_additive additive_pos "
+  Let `α` be a lattice ordered commutative group with identity `0`. For an element `a` of type `α`,
+  the element `a ⊔ 0` is said to be the *positive component* of `a`, denoted `a⁺`.
+"]
```

Comment:
> `additive_pos` is the notion that people actually care about, right? The multiplicative version of this story is just there because we _can_ do it.

It's the one I'm interested in - but I can't speak for other people. It makes sense in the context of vector lattices.

Reply:
There isn't an established name for "multiplicatively positive". So I suggest that we call it `mpos`. Similarly `mneg`. Then the additive variants can be called `pos` and `neg`.

Reply:
Done. (Most of the time we use the ⁺ and ⁻ postfix rather than the names anyway.)